### ISSUE TITLE
Expected URL scheme 'http' or 'https' but was 'asset'

### ISSUE THREAD

**Issue by spacens** [expo/expo#8257]
### Environment

Expo CLI 3.15.5 environment info:
    System:
      OS: macOS 10.15.2
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 10.13.0 - ~/.nvm/versions/node/v10.13.0/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.4.1 - ~/.nvm/versions/node/v10.13.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6392135
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^35.0.0 => 35.0.1 
      react: 16.8.3 => 16.8.3 
      react-native: https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz => 0.59.8 
      react-navigation: ^3.11.0 => 3.13.0 

The app is built using expo and then ejected with ExpoKit. It runs perfectly on iOS and in Android debug mode but I am getting errors in Android release mode.

### Steps to Reproduce

Create a new expo project(v35) and eject using ExpoKit. I am using many packages in the project but I think that this specific issue is related to the splash screen though I am not 100% sure. Following is a part of the configuration in build.gradle.
```
implementation "com.android.support:exifinterface:${safeExtGet("supportLibVersion", "28.0.0")}"
implementation 'com.squareup.okio:okio:1.9.0'
implementation 'com.facebook.soloader:soloader:0.6.0'

// expo-file-system
implementation 'com.squareup.okhttp3:okhttp:4.6.0'
implementation 'com.squareup.okhttp3:okhttp-urlconnection:4.6.0'
```

### Expected Behavior

The release apk should run without issues.

### Actual Behavior

I debugged the apk from Android Studio and got the following errors:
```
E/GlideExecutor: Request threw uncaught throwable
    java.lang.IllegalArgumentException: Expected URL scheme 'http' or 'https' but was 'asset'
        at okhttp3.HttpUrl$Builder.parse$okhttp(HttpUrl.kt:10)
        at okhttp3.HttpUrl$Companion.get(HttpUrl.kt:1)
        at okhttp3.Request$Builder.url(Request.kt:6)
        at com.bumptech.glide.integration.okhttp3.a.a(OkHttpStreamFetcher.java:1)
        at com.bumptech.glide.load.b.L.a(SourceGenerator.java:12)
        at com.bumptech.glide.load.b.l.q(DecodeJob.java:4)
        at com.bumptech.glide.load.b.l.r(DecodeJob.java:4)
        at com.bumptech.glide.load.b.l.run(DecodeJob.java:7)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
        at com.bumptech.glide.load.b.c.a.run(GlideExecutor.java:8)
```

I don't know if this is an issue with okhttp or expo. I have been struggling with this issue for 2 days already.

Thank you in advance.

**Comment by brentvatne** [comment]
hi! we really strongly recommend updating to sdk37 and then ejecting to bare workflow. expokit is deprecated and you will need to move away from it eventually

**Comment by spacens** [comment]
@brentvatne do you think that it will solve the issue?

**Comment by cruzach** [comment]
@spacens there's a good chance it does, yes

**Comment by spacens** [comment]
@brentvatne I have created a brand new project using sdk37 and ejected to bare workflow. I copied the src folder from the original project and now I am getting a lot more errors. I even can't get it running on iOS(It was working perfectly on iOS with sdk35).
Does it mean that I have to refactor the entire app?
I wouldn't do it because it 's currently working fine on iOS with sdk35. I just have the "assets" issue with android release version.
I have spent another week on this issue but couldn't find any good migration guide.
